src/Controller.php

```diff
@@ -34,7 +34,13 @@ private function getProvider($provider, $redirectUri = null)
     {
         $config = $this->getConfig($provider) + compact('redirectUri');
 
-        $class  = 'League\\OAuth2\\Client\\Provider\\' . ucfirst($provider);
+        if (isset($_ENV[strtoupper($provider) . '_NAMESPACE'])) {
+            $namespace = $_ENV[strtoupper($provider) . '_NAMESPACE'];
```

Comment:
It's not perfect but not really an inconvenience either. When adding another provided just remember to import the namespace. Config only would be awesome but the namespaces aren't consistent enough for it to be a realistic expectation.